grammar.y

```diff
@@ -1212,6 +1219,75 @@ insert_rows:
   }
 ;
 
+upsert_clause_opt:
+  {
+    $$ = nil
+  }
+| on_conflict_clause_list
+  {
+    allConflictClausesExceptLast := $1[0:len($1) - 1];
+    for _, clause := range allConflictClausesExceptLast {
+      if clause.Target == nil {
+        yylex.(*Lexer).AddError(&ErrUpsertMissingTarget{})
+      }
+    }
```

Comment:
Reference from the spec to know why this is the case:
> The conflict target may be omitted on the last ON CONFLICT clause in the INSERT statement, but is required for all other ON CONFLICT clauses.

NRE, just helping if anyone else is reading this.